mindsdb/integrations/handlers/eventstoredb_handler/eventstoredb_handler.py

```diff
@@ -0,0 +1,212 @@
+    host = 'localhost'
+    port = '2113'
+    is_connected = None
+    basic_url = ""
+    read_batch_size = 500  # should be adjusted based on use case
+    headers = {
+        'Accept': 'application/json',
+        'ES-ResolveLinkTo': "True"
+    }
+    tlsverify = False
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name)
+        self.parser = parse_sql
+        connection_data = kwargs['connection_data']
+        self.host = connection_data.get('host')
+        if connection_data.get('tls') is not None and isinstance(connection_data.get('tls'), bool) \
+                and connection_data.get('tls'):
+            self.scheme = 'https'
+        if connection_data.get('port') is not None and isinstance(connection_data.get('port'), int):
+            self.port = connection_data.get('port')
+        if connection_data.get('page_size') is not None:
+            if isinstance(connection_data.get('page_size'), int) and connection_data.get('page_size') > 0:
+                self.read_batch_size = connection_data.get('page_size')
+        log.logger.debug(f'scheme: {self.scheme} host: {self.host}  port:{self.port} page: {self.read_batch_size}')
+        if connection_data.get('username') is not None and connection_data.get('password') is not None:
+            if isinstance(connection_data.get('username'), str) and isinstance(connection_data.get('password'), str):
+                self.headers['authorization'] = get_auth_string(connection_data.get('username'),
+                                                                connection_data.get('password'))
+        if connection_data.get('tlsverify') is not None and isinstance(connection_data.get('tlsverify'), bool):
+            self.tlsverify = connection_data.get('tlsverify')
+
+        self.basic_url = build_basic_url(self.scheme, self.host, self.port)
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self, **kwargs):
+        if self.check_connection() == StatusResponse(True):
+            self.is_connected = True
+            log.logger.info(f'{self.name} connection successful!')
+            return StatusResponse(True)
+        log.logger.info(f'{self.name} connection could not be made.')
+        return StatusResponse(False)
+
+    def disconnect(self, **kwargs):
+        return
+
+    def check_connection(self) -> StatusResponse:
+        try:
+            response = requests.get(build_health_url(self.basic_url), verify=self.tlsverify)
+            if response.status_code == 204:
+                return StatusResponse(True)
+        except Exception as e:
+            log.logger.error(f'{self.name} check connection failed with: {e}!')
+        return StatusResponse(False)
+
+    def query(self, query: ASTNode) -> Response:
+        if type(query) == Select:
+            stream_name = query.from_table.parts[-1]  # i.e. table name
+            params = {
+                'embed': 'tryharder'

```

Comment:
If this can be more then one value, should we add option for setting it up in the query?

Reply:
Do you mean params? At the moment we do not need to add any additional parameters or change it. Do you want me do some refactor on it?
Thanks for the prompt review!